### ISSUE TITLE
Aggregate function that returns a type that cannot be Nullable should still be supported for `aggregate_functions_null_for_empty`

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#38786]
@Alex-Cheng these functions should return the default value instead of NULL.

Now it's not the case:
```
SELECT groupArray(x) FROM (SELECT toNullable(1) AS x) SETTINGS aggregate_functions_null_for_empty = 1

Code: 43. DB::Exception: Nested type Array(UInt8) cannot be inside Nullable type. (ILLEGAL_TYPE_OF_ARGUMENT)
```

And should be improved.

_Originally posted by @alexey-milovidov in https://github.com/ClickHouse/ClickHouse/issues/10273#issuecomment-1173333688_

**Comment by Alex-Cheng** [comment]
@alexey-milovidov  Yes, I agree with you. The problem is that the aggregate_functions_null_for_empty = 1 settings ALWAYS add -OrNull suffix to all aggregate functions including `groupArray` function, but groupArrayOrNull does not work at all because of the reason you mentioned. It is the issue at version 21.12.3.1, not sure if the issue has been fixed in later version.

BTW, I am glad to hear from you and I've been looking forward to meet you for a long time – I'm honored!
